docs: Choosing the Right Prompting Style Blog (#1794)
block/goose: documentation/blog/2025-03-19-prompts/index.md

---

> title: "Stop Confusing Your AI Agent"

Reviewer: you need a more seo-friendly title. something about prompting in AI

description: How to prompt your AI agent the right way and vibe code your way to better responses.
authors: 
    - ebony
---

> Remember that saying, It’s not what you ask, but **how you ask**?

Reviewer: ```suggestion
Remember that saying, "it’s not what you ask, but **how you ask**"?
```

> When I first started working with Goose and AI agents, I was convinced there had to be one ‘best' prompting style. I spent so much time trying to figure out which one was superior, but the more I used Goose, the more I realized that couldn't be further from the truth. There isn’t one ‘right’ way to prompt AI, but there are better approaches depending on what your end goal is.

Reviewer: ```suggestion
When I first started working with Goose as an AI agent, I was convinced there had to be one ‘best' prompting style. I spent so much time trying to figure out which one was superior, but the more I used Goose, the more I realized that couldn't be further from the truth. There isn’t one _right_ way to prompt AI, but there are better approaches depending on what your end goal is.
```

So, let’s go through **which prompt style works best for your specific needs**, and how you can use them to vibe code a little better with Goose.

<!--truncate-->

## Instruction-Based Prompting

If you’re not a developer or new to Goose, this is a great place to start. The best way to get good responses is to be as clear and direct as possible. Goose works best when given specific instructions, so tell it exactly what you need and include all of the important details. If you’re too vague, you might end up with an overly technical or even a possibly incomplete answer that doesn’t actually help you.

For example, instead of saying:

❌ Okay Prompt: 

>_**Goose, what’s a pull request?**_ 

This might give you a super technical definition that assumes you already know the basics. 

So, you could say:

✅ Better Prompt:
>_**Goose, explain how GitHub pull requests work like I’m new to coding**_ 

This tells Goose exactly what you need and at what level. 

:::tip pro tip

> If you want Goose to remember your preferences, you can simply say, 

Reviewer: ```suggestion
If you want Goose to remember your preferences, you can say, 
```

>_**Goose, remember I’m not a developer. Explain things at a high level unless I ask for technical details**_

If you have the [Memory Extension](/docs/tutorials/memory-mcp) enabled, Goose will save this preference so you won’t have to remind it every time. 
:::

> But what if you’re tackling something more complex that needs a step-by-step breakdown? That’s where Chain-of-Thought Prompting comes in.

Reviewer: ```suggestion

```

redundant
